### DISCUSSION TITLE
ducted heat pump system

### DISCUSSION THREAD

**Discussion by skyw33** [echavet/MitsubishiCN105ESPHome#105] [Show and tell]
USA

Mitsubishi Ducted Heat Pump System

OUTDOOR unit:  PUZ-HA42NKA1
INDOOR unit:  PVA-A42AA7

Thermostat:  Mitsubishi MHK2 Wireless Remote Controller

Airzone CN105 WiFi Splitter
https://gotductless.com/products/airzone-aidoo-cn105-splitter-for-mitsubishi

I use the MHK2 as my thermostat and then I use ESPHome to control the unit from home assistant. https://github.com/echavet/MitsubishiCN105ESPHome

Both the MHK2 and the ESPHome unit connect to the splitter, while the splitter connects to the CN105.

Theoretically, I could get rid of the Wifi Splitter and MHK2, but I'm hesitant to do so because my understanding is that the MHK2 has smarts that are better able to control the mitsubishi heat pump.

I'm using these ESP modules:
ESP01 adapter:  https://www.amazon.com/dp/B09S11Q4TL?psc=1&ref=ppx_yo2ov_dt_b_product_details
ESP8266 ESP01 module: https://www.amazon.com/dp/B01EA3UJJ4?ref=ppx_yo2ov_dt_b_product_details

And this CN105 cable:
https://www.aliexpress.us/item/3256802718583041.html?spm=a2g0o.order_list.order_list_main.17.ced11802eB8p2O&gatewayAdapt=glo2usa

In the picture below you see the wifi splitter with 3 connections.  From left to right: cable to CN105, cable to ESP8266, cable to MHK2.

Note for the ESP8266 cable that the purple wire is not used.  Green=GND, Yellow=VCC, Red=TX, Black=RX
![PXL_20240524_145427173 MP](https://github.com/echavet/MitsubishiCN105ESPHome/assets/26240042/3e289903-e91b-4c54-875a-f2ad5c3e8ce3)

**Comment by skyw33** [comment]
```
esphome:
  name: heatpump

esp8266:
  board: esp01_1m



uart:
  id: HP_UART
  baud_rate: 2400
  tx_pin: 1
  rx_pin: 3



# Enable Home Assistant API
api:
  password: ""

ota:
  password: ""

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "Heatpump Fallback Hotspot"
    password: "Qv4rOYLVOtYK"

captive_portal:

substitutions:
  name: heatpump-1 # Do not use underscores, which are not fully compatible with mDNS
  friendly_name: My Heatpump 1


# External component reference
external_components:
  - source: github://echavet/MitsubishiCN105ESPHome

climate:
  - platform: cn105
    id: hp
    name: "${friendly_name}"
    icon: mdi:heat-pump
    #visual:
    #  min_temperature: 15
    #  max_temperature: 31
    #  temperature_step:
    #    target_temperature: 1
    #    current_temperature: 0.1
    compressor_frequency_sensor:
      name: ${name} Compressor Frequency
    #vertical_vane_select:
    #  name: ${name} Vertical Vane
    #horizontal_vane_select:
    #  name: ${name} Horizontal Vane
    isee_sensor:
      name: ${name} ISEE Sensor
    #remote_temperature_timeout: 30min
    #debounce_delay : 500ms
    update_interval: 4s
    stage_sensor:
      name: Stage Sensor
    sub_mode_sensor:
      name: Sub Mode Sensor
    auto_sub_mode_sensor:
      name: Auto Sub Mode Sensor



logger:
  hardware_uart: UART1
  level: INFO
  baud_rate: 0
  logs:
    EVT_SETS : INFO
    WIFI : INFO
    MQTT : INFO
    WRITE_SETTINGS : INFO
    SETTINGS : INFO
    STATUS : INFO
    CN105Climate: WARN
    CN105: INFO
    climate: WARN
    sensor: WARN
    chkSum : INFO
    WRITE : WARN
    READ : WARN
    Header: INFO
    Decoder : INFO
    CONTROL_WANTED_SETTINGS: INFO

web_server:
  port: 80

time:
  - platform: homeassistant
    id: homeassistant_time

text_sensor:
  # Expose ESPHome version as sensor.
  - platform: version
    name: ${name} ESPHome Version
  # Expose WiFi information as sensors.
  - platform: wifi_info
    ip_address:
      name: ${name} IP
    ssid:
      name: ${name} SSID
    bssid:
      name: ${name} BSSID


button:
  - platform: restart
    name: "Restart ${friendly_name}"
```

**Reply by akifbayram** [reply]
The ESPHome unit _should_ have almost the same smarts as the MHK1/MHK2 since it uses the Mitsubishi protocol. A third-party thermostat (like ecobee, nest, etc.) would only report that it needs to heat or cool (maybe  multi-stage?).

**Reply by skyw33** [reply]
Well, it is very tempting to think that I can replace everything with this ESP device, but I've invested quite a bit of money in this system and I'm hesitant to experiment.

**Reply by skyw33** [reply]
Here's an example of things that can be done with mhk2 that are not possible any other way (correct me if I'm wrong):  https://www.mylinkdrive.com/viewPdf?srcUrl=http://s3.amazonaws.com/enter.mehvac.com/DAMRoot/Original/10006%5CApplication%20Note%203048%20ME%20-%20How%20to%20turn%20off%20indoor%20unit%20fan%20when%20set%20point%20is%20met.pdf

**Reply by akifbayram** [reply]
Good point. I keep my MHK1 in storage in case I need to change/set the thermal off fan operation. Fortunately, ever since I set it to 'Off', I haven't needed to re-set it. Seems like the function itself has been [decoded](https://github.com/SwiCago/HeatPump/issues/39#issuecomment-1040924886) by SwiCago.

**Reply by skyw33** [reply]
Amazing that they have that decoded! So impressed by all the work going into this project and so happy with the results 😊

**Reply by skyw33** [reply]
@akifbayram @echavet What do you think the chances are of incorporating this functionality from swicago into this project?  I.e. the ability to change these functions from within home assistant would be very helpful and allow me to get rid of my mhk2 thermostat.  In particular, I would love the ability to change settings 125 and 127 more easily.
(see screenshots).
![Screenshot from 2024-06-07 10-12-07](https://github.com/echavet/MitsubishiCN105ESPHome/assets/26240042/c9408272-6ec6-4b03-a9cd-6c1fa4404a0c)
![Screenshot from 2024-06-07 10-15-03](https://github.com/echavet/MitsubishiCN105ESPHome/assets/26240042/1f32edd5-eff1-415c-9d18-cc930523c38c)

**Reply by mrmattyq** [reply]
@skyw33 sorry to resurrect an old discussion.. but.. I'm trying to accomplish a similar setup.

HP -> Airzone Splitter -> ESP & MHK2.

I have a near identical config, but different board (MakerFocus ESP8266 ESP-12F N). When i plug the ESP directly into the HP everything works, but when i plug into the splitter it doesn't. My Outdoor & Indoor units are also different (Mini split MSZ-FS12). Did you experience any similar behavior in your testing? 

I know not all boards are created equal, so I've ordered the parts you listed which hopefully fix's this "issue". 

I guess my question is, how is the function parity between ESP + MKH2. For example, if you set the setpoint in HA, does it sync to thermostat and vice versa? I want to use this instead of buying an Airzone Aidoo - but that may be the route i have to go if i cant get it to work!

**Reply by skyw33** [reply]
For more context, I started with MHK2 + Airzone splitter + Airzone Aidoo -- this was my baseline.  After that was all working, I ditched the Aidoo and now use ESP.  Therefore, it's possible that setting up the Aidoo app somehow primed the splitter to work properly???  Just speculating...  Anyway, I never had any trouble connecting ESP to the splitter -- it just worked.  And the functionality is just what you describe -- you can control the HP from either the MHK2 or ESP and it will sync/recognize that values were changed on each.  I've never tried to connect ESP directly to the CN105 port, so I don't have any additional information about that.

**Reply by mrmattyq** [reply]
Interestingly enough, switching to the ESP8266 ESP01 + adapter worked!

**Reply by skyw33** [reply]
Nice!  Good to hear :)

**Reply by dsstewa** [reply]
I am running into the same issue as above, Wemos D1 mini and ESP 32 do not work through the splitter. Looks like I will be ordering an ESP8266 and ESP01 as noted above. I wonder if it has to do with a resister....

**Reply by theflu** [reply]
You might need a pull up resistor on the tx/rx lines. As noted in the SwiCago/HeatPump repo.
https://github.com/SwiCago/HeatPump?tab=readme-ov-file#demo-circuit

**Comment by skyw33** [comment]
![Screenshot from 2024-05-24 18-03-52](https://github.com/echavet/MitsubishiCN105ESPHome/assets/26240042/f2764403-5ae9-4349-b7ee-9784bfc30ad1)
![Screenshot from 2024-05-24 18-04-16](https://github.com/echavet/MitsubishiCN105ESPHome/assets/26240042/e10496c1-1e7b-4b56-91dc-718ad06e0602)
![Screenshot from 2024-05-24 18-04-46](https://github.com/echavet/MitsubishiCN105ESPHome/assets/26240042/f3f55eb7-d2ee-4fec-9745-f657db4c6a36)

**Comment by tmchow** [comment]
The linked ESp8266 unit is no longer available. Will any equivalent unit work like

https://amzn.to/3Yp8m9b
or
https://amzn.to/3ylxAL8
Or
https://amzn.to/3Wtx3yP

?

Also I’m looking for a different sourde than AliExpress. So life of me i cannot get a transaction to go through due to their fraud detection and their support is of no help 🤦‍♂️

**Comment by tmchow** [comment]
@mrmattyq @skyw33 

can you share the pin out configuration from the ACC port on the Airzone splitter? It’ll be helpful to know it exactly.  I couldn’t make out fully the wire colors from the photo in the original post. 

Update: Also, can you confirm if you’re using the AZ port or the ACC port? I realize after I posted my original question about the Pinout I wasn’t sure. Currently I have the MKH2 red link receiver connected to AZ but in looking at your photo is seems you have it reversed

**Reply by mrmattyq** [reply]
This is the board I used which still shows available:

https://amzn.to/3ylxAL8.

I posted my working config [here](https://github.com/SwiCago/HeatPump/issues/13#issuecomment-2228886009) - hope that helps!

**Reply by echavet** [reply]
Here is a nice feedback with the atom chip:
https://github.com/echavet/MitsubishiCN105ESPHome/discussions/83

**Reply by tmchow** [reply]
@echavet Yeah I'm following that one but having issue with the Airzone CN105 connection as it doesn't seem to be powering up the Atom S3 Lite. This is why I'm asking about the exact pin mapping to the AZ port.  While @mrmattyq used a different board, I could make it work with the Atom s3 if i could verify the pin configuration.

**Reply by echavet** [reply]
I don't understand what you mean by pin configuration.
Pin configuration is the same for all CN105 devices, no?
What is missing in the #83 discussion ?

**Reply by tmchow** [reply]
Is the AZ and ACC port on the Airzone CN105 ports exactly the same pin configuration? If so, then I'm so confused because my wiring does not power up when it's plugged into the Airzone CN105 but will turn on when i plug in directly to the Mitsubishi unit.

**Reply by echavet** [reply]
Well I thought they were the same, but I only have standard units. Never tested anything else.

**Reply by tmchow** [reply]
Yeah this is why I'm asking to be sure. If you know the pin out mapping on the ESP01 unit you have, you could just tell me what colors they map too on the Airzone CN105 connector 🤞

**Reply by echavet** [reply]
I wish I could help but i don't have any airzone device.
I don't even understand the interest of such device. If all of your units are controlled through an esp connected to home assistant what is the point of using an airzone ?

**Reply by tmchow** [reply]
The advantage is to have failsafe of act actual thermostat device in case there's ever something wrong with Home Assistant or the ESP device.  HVAC is one of those things that has to be available to control at all times.

I have the Airzone CN105 connected to my main system and the MKH2 connected to it. ESP device connected to the other plug of the airzone.

**Comment by tmchow** [comment]
@skyw33 @mrmattyq
Ok I've given up on the ATOM S3 Lite, Atom S3 and the Wemos D1 Mini. None of them work with the Airzone CN105 splitter it seems. The ATOMs won't even power up when connected, and the Wemos D1 mini can't seem to communicate to it, which i think is related to the power for reading from serial that others have reported.

I've been fighting buying the ESP01 modules, but looks like I have no choice now :)

Do I need both these things? I thought ESP01 was a wifi module based on ESP8266 SoC, but it appears the guidance is to connect these things together but i may be misunderstanding:

- [ESP8266 ESP01 module](https://amzn.to/3YQEUcC)
- [ESP01 Adapter](https://amzn.to/4fQtUBV)

Assuming it's correct I need both those things, the bew newbie question here is how I connect via usb to these boards at all?  With my ATOM S3 and Wemos D1 there was a usb cable.  Do I need to buy one of these [ESP-01 USB programmers](https://amzn.to/3AD1TgV)?

**Reply by skyw33** [reply]
All of your assumptions are correct.  First you use the programmer to get the firmware on the esp01 module, then you connect it to the adapter, and finally, connect the adapter to the splitter with the CN105 cable (https://www.aliexpress.us/item/3256802718583041.html?spm=a2g0o.order_list.order_list_main.17.ced11802eB8p2O&gatewayAdapt=glo2usa)

Happy to help if you have any questions along the way.

**Reply by dsstewa** [reply]
The ESP-01 are the way to go, I changed out all of my Wemos D1 mini's. It's a much cleaner install and as you have found out is the only thing that works with the airzone controller.![image](https://github.com/user-attachments/assets/c1496112-a90d-482e-aaca-9f23b46b665e) ![image](https://github.com/user-attachments/assets/0faa38eb-fcd2-4c7b-82ea-163511c1fb96)

**Comment by tmchow** [comment]
THANK YOU to everyone who commented in this thread. Finally got all my parts in and used the ESP-01 like everyone else mentioned. It was the only one I could get to work with the Airzone CN105. I now have it working to control via Home Assistant through ESPHome, and also have an MKH2 Thermostat as a fallback (a) in case HA is every down for some reason or (b) someone in my household just wanted something physical to use.

**Comment by bass-windu** [comment]
Works with my multi-split-system composed of a ducted air handler (SVZ-KP30NA) + wall mounted mini split (MSZ-FS09NA) on one 48k BTU outdoor unit (MXZ-SM48NAMHZ2)

I too am using an airzone splitter and ESP01 and my only issue is the current temperature and desired temperature on my MHK2 is off by 1 degree F from home assistant. I am not using remote temperature on home assistant.

**Reply by phidauex** [reply]
The temperature being off is the result of Mitsubishi using their own C to F conversion table that usually, but doesn't always, match the actual math. This means that in the double conversion of F to C to F, HA can get one off from the thermostat. This isn't a big problem, but it is annoying. The mitsubishi2mqtt folks have solved it with a conversion table, as shown in this PR (https://github.com/gysmo38/mitsubishi2MQTT/pull/272), but I'm not sure that the solution will work for us here since some of the conversion is happening in Home Assistant.

**Reply by bass-windu** [reply]
Ok wonderful. I will not worry about it. Cheers!

**Comment by jharris4** [comment]
I was tearing my hair out after trying several models of esp32 trying to get them to work with the airzone splitter, and thanks to finding this thread I was able to successfully get it working through the splitter with an esp01. Many thanks to everyone here!

One part that really tripped me up with the esp01 is that it had trouble connecting to wifi whenever I had the climate component in the firmware by adding it to the yaml. If I went back to a basic yaml file without the climate component, the esp01 connected to wifi just fine. Quite the head scratcher, but by luck I figured out that I needed to enable `fast_connect` and specify `manual_ip` in the `wifi` section of the yaml and then everything worked.

**Comment by mgulick** [comment]
For the ducted units with an airzone splitter, the problem with the ESP32 seems to be the 3.3V logic levels.  You can still use them, but you'll need to build your own board with a level shifter to get the logic levels back up to 5V.  I'm using this configuration with two ESP32-S3 boards.  See the discussion here: https://github.com/echavet/MitsubishiCN105ESPHome/discussions/273

**Comment by jharris4** [comment]
@mgulick thanks for following up on this!

Last week I was doing some network upgrades and it turns out the esp01 crashes when it can’t find wifi.

it also turns out hvac goes nuts when the esp01 is unresponsive.

it took me a couple of hours to notice that the heating had been running nonstop… 😬

Suffice it to say I’m hoping that getting an actual esp32 working with the air zone splitter resolves some of these issues!
